src/api/course/index.ts

```diff
@@ -137,10 +137,84 @@ export class CourseApi {
           this.processV1DestinationDeltas(msg)
         }
       )
+      ;(this.app as any).subscriptionmanager.subscribe(
+        {
+          context: 'vessels.self',
+          subscribe: [
+            {
+              path: 'resources.routes.*',
+              period: 500
+            },
+            {
+              path: 'resources.waypoints.*',
+              period: 500
+            }
+          ]
+        },
+        this.unsubscribes,
+        (err: Error) => {
+          console.log(`Course API: Subscribe failed: ${err}`)
+        },
+        (msg: Delta) => {
+          this.processResourceDeltas(msg)
+        }
+      )
       resolve()
     })
   }
 
+  /**
+   * Resource delta message processing to ensure update made to
+   * 1. waypoint referenced as the current destination OR
+   * 2. active route
+   * are reflected in course deltas.
+   */
+  private processResourceDeltas(delta: Delta) {
+    let h: string[]
+    if (this.courseInfo.activeRoute?.href) {
+      h = this.courseInfo.activeRoute?.href.split('/').slice(-3)
+    } else if (this.courseInfo.nextPoint?.href) {
+      h = this.courseInfo.nextPoint?.href.split('/').slice(-3)
+    } else {
+      return
+    }
+    const ref = h ? h.join('.') : undefined
+    const refType = h ? h[1] : undefined
+
+    delta.updates.forEach((update: Update) => {
+      if (hasValues(update)) {
+        update.values.forEach(async (pathValue: PathValue) => {
+          if (ref === pathValue.path) {
+            if (refType === 'routes') {
+              const rte = await this.getRoute(
+                (this.courseInfo.activeRoute as ActiveRoute).href
+              )
+              if (rte) {
+                ;(this.courseInfo.activeRoute as ActiveRoute).name =
+                  rte.name as string
```

Comment:
Same question here. Is this because `ActiveRoute.name` can be `null` and `Route.name` can't? I would advocate for either:

1. updating the types to make the compatible, or
2. add guards to handle null/undefined
3. as a last resort, use `rte.name!` if you are certain it's not null (or ok that it is null), but again this is usually a smell of incompatible or insufficient interfaces/types.